src/client/mvt/feature-decoder.js

```diff
@@ -85,24 +105,32 @@ export function clipPolygon(preClippedVertices, polygon, isHole) {
 
             if (insideA && insideB) {
                 // case 1: both inside, push B vertex
+                setClippedType(preClippedVertices2.length, k + 1);
                 preClippedVertices2.push(b);
             } else if (insideA) {
                 // case 2: just B outside, push intersection
                 const intersectionPoint = clippingEdgeIntersectFn[i](a, b);
+                setClippedType(preClippedVertices2.length, k + 1, i);
                 preClippedVertices2.push(intersectionPoint);
             } else if (insideB) {
                 // case 4: just A outside: push intersection, push B
                 const intersectionPoint = clippingEdgeIntersectFn[i](a, b);
+                setClippedType(preClippedVertices2.length, k, i);
                 preClippedVertices2.push(intersectionPoint);
+                setClippedType(preClippedVertices2.length, k + 1);
                 preClippedVertices2.push(b);
             } else {
                 // case 3: both outside: do nothing
             }
         }
         if (preClippedVertices2.length) {
+            if (clippedTypes2[0]) {
+                clippedTypes2[preClippedVertices2.length] = clippedTypes2[0];
+            }
             preClippedVertices2.push(preClippedVertices2[0]);
         }
         preClippedVertices = preClippedVertices2;
+        clippedTypes = clippedTypes2;
     }
 
     if (preClippedVertices.length > 3) {
```

Comment:
Why 3?

Reply:
With less than 3 vertices we have a degenerate ring (a line or a point), not a proper one, so we ignore it.

Reply:
Hey, that was wrong! I think it shoud be >= 3

Reply:
... unless we're accounting for the first vertex being repeated

Reply:
Yes, that's right 😅 when we get there we have duplicated the first vertex at the end to close the ring. But if we end up with 3 or less vertices (so we have at most 2 distinct vertices) that must be a degenerate case and we omit it

Reply:
Ok, I would use then `const MAX_VERTICES = 3` to make it more understandable :)